rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -880,6 +891,12 @@ extern "C" rmw_ret_t rmw_init(const rmw_init_options_t * options, rmw_context_t
     return RMW_RET_BAD_ALLOC;
   }
 
+  /* "impl"'s destructor relies on these being initialized properly */
+  impl->common.thread_is_running.store(false);
+  impl->common.graph_guard_condition = nullptr;
+  impl->common.pub = nullptr;
+  impl->common.sub = nullptr;
```

Comment:
can these be initialized in `rmw_context_impl_t` constructor?

Reply:
Yes, that'd be better, thanks